businessRules/donations.js

```diff
@@ -1,15 +1,17 @@
 const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
-const { getAllIdeas } = require('../models/ideas');
-const { getIdeas } = require('../utils/storage');
+const {
+    ideaThatMatchesIdeaId,
+} = require('../validation/ideaFunctionValidation');
+const {
+    checkIfSumIsANumber,
+    ideaWithTotalDonationSumThatMatchesIdeaId,
+} = require('../validation/donationFunctionValidation');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createDonationsWithDataValidation(firstName, sum, userId, ideaId) {
```

Comment:
This `createDonationsWithDataValidation` function is a public API of this module. The name `createDonationsWithDataValidation` tells the world that this function not on;y creates a donation, but also validates the data - is there a reason why the world needs to know about data validation?

Reply:
I think I've fixed it.